### PR TITLE
Serialize accessor to csv, pickle, and parquet

### LINKED ISSUE #560: Implement serialization to WoodworkTableAccessor
- Adds `WoodworkTableAccessor.to_dictionary`, `WoodworkTableAccessor.to_csv`, `WoodworkTableAccessor.to_pickle`, `WoodworkTableAccessor.to_parquet`
- Will need to create separate seralization files until the DataTable class is removed
- Will need to update Serialization schema version and add new file to S3

This issue implements the serialization method.

### PR DESCRIPTION
**Pull request by tamargrey**
- creates `serialize_accessor.py` file for serializing a DataFrame and its Woodwork Schema and `test_accessor_serialization.py` 
- Closes #560 

Note: Most of the serialization tests require the ability to deserialize as well, so those tests will have to wait until deserialization is implemented

### CODE REVIEW

**Comment by tamargrey** [comment]
The `serialize_accessor.py` file is rife with opportunities for bad name overlap with our `Schema` class and the `schema version`, `table_metadata` vs `column_metadata` being attributes of typing information but previously being used to describe the serialized typing information, so I've tried to stick to the following naming conventions in the file:

- "schema" - refers to the version used to serialize
- "typing information"/"typing info" - the typing that Woodwork provides for a table
- "column_typing_info" - the typing information that Woodwork provides organized by column
- "Woodwork table" - the combination of DataFrame + Typing Information

File: woodwork/serialize_accessor.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    '''
+    ordered_columns = dataframe.columns
+    column_typing_info = [
+        {'name': col_name,
+         'ordinal': ordered_columns.get_loc(col_name),
+         'logical_type': {
+             'parameters': _get_specified_ltype_params(col['logical_type']),
+             'type': str(_get_ltype_class(col['logical_type']))
+         },
+            'physical_type': {
```
Comment (thehomebrewnerd) [inline woodwork/serialize_accessor.py:36]: Extra indent here

File: woodwork/serialize_accessor.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    elif _is_url(path):
+        raise ValueError("Writing to URLs is not supported")
+    else:
+        path = os.path.abspath(path)
+        os.makedirs(os.path.join(path, 'data'), exist_ok=True)
+        dump_table(dataframe, path, **kwargs)
+
+
+def dump_table(dataframe, path, **kwargs):
+    '''Writes Woodwork table the specified path, including both the data and the typing information.
```
Comment (thehomebrewnerd) [inline woodwork/serialize_accessor.py:93]: Missing word -> Writes Woodwork table **at** the specified path.

**Review by thehomebrewnerd — APPROVED** [review]
Just a couple minor wording issues, otherwise I didn't see anything that needs to be updated, recognizing we might have to fix some serialization steps once we have implemented deserialization to complete the tests.

File: woodwork/serialize_accessor.py (hunk 1 above)
Comment (tamargrey) [reply woodwork/serialize_accessor.py]: unindented this and the couple of fields below that are also too far indented

**Pull request merged**